config/default-config.yml

```diff
@@ -36,6 +36,8 @@ sites:
     skip_provisioning: false
     description: "A standard WP install, useful for building plugins, testing things, etc"
     repo: https://github.com/Varying-Vagrant-Vagrants/custom-site-template.git
+    php: 7.4 # change the PHP version to use for the provision
+    nginx_upstream: 7.4 # change the php version to use for nginx
```

Comment:
the hope is we can remove the nginx upstream because PHP was set to 7.4, as a result the provisioner can infer what the upstream value should be

Reply:
Right now they are 2 different values but I am not sure what this feedback is just a note or something to do

Reply:
I don't think we should mention the nginx upstream here, and it should work without it and just `php: 7.4`

Reply:
So in the future the docs would say just put `php: 8.0` and it would run the provisioners in 8.0 and the nginx config would also use PHP 8 instead of needing an nginx upstream value.

Then in a future release we can auto-run the PHP 8 extension before the sites provision based on what the config says

Reply:
So with a single value change both the parameters.
Maybe it's the case to open a ticket just for this.

Reply:
but we did open a ticket for this, https://github.com/Varying-Vagrant-Vagrants/VVV/issues/2367 which is what this PR is meant to implement

Reply:
This is still a hard blocker and a major issue, it implies a misunderstanding of what #2367 was asking for. The whole point of the issue is that you no longer need to specify an Nginx upstream to change your PHP version, you only specify an upstream if you specifically and only want to change the upstream, not because you want to change the PHP version used.

Reply:
The problem with this is: what if I want that the provision is executed with a php version and nginx with another one?

Reply:
then it would use the nginx upstream in the config because the user has explicitly chosen the value.

 - `php` tells VVV which version of PHP to use for the site
 - `nginx_upstream` allows you to override a value in the nginx config

Right now we use `nginx_upstream` to set the version of PHP used on frontend requests because no other options exist, it's a workaround. This issue is meant to fix that by providing a `php:` parameter